### PR TITLE
Basic Link Handling

### PR DESCRIPTION
**Pull request by k-e-l-p**

### CODE REVIEW

File: plugins/link-handler.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	var output string
+
+	// NOTE(nojusr): in PRIVMSG's case, the second (first, if counting from 0) parameter
+	// is the string that contains the complete message.
+	for _, value := range strings.Split(m.Params[1], " ") {
+		u, err := url.Parse(value)
+
+		if err != nil {
+			continue
+		}
```
Comment (icyphox) [inline plugins/link-handler.go:45]: bruh

File: plugins/link-handler.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		// NOTE(nojusr): just a test check for the time being.
+		// this if statement block will be used for content that is
+		// non-generic. I.e it belongs to a specific website, like
+		// stackoverflow or youtube.
+		if u.Hostname() == "youtube.com" || u.Hostname() == "youtu.be" {
+			// TODO(nojusr): finish this
+			output += "[Youtube] yeah you definitely posted a youtube link"
+		} else if len(u.Hostname()) > 0 {
+			output += "[URL] "
+			output += getDescriptionFromURL(value)
```
Comment (icyphox) [inline plugins/link-handler.go]: probably use `strings.Builder` for this? or `fmt.Sprintf` works too.

File: plugins/plugins.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ func ProcessTrigger(m *irc.Message) (string, error) {
 		err      error
 	)
 	for trigger, plugin := range Plugins {
-		if strings.HasPrefix(m.Trailing(), trigger) {
+		if strings.HasPrefix(m.Trailing(), trigger) || trigger == "" {
```
Comment (icyphox) [inline plugins/plugins.go]: is there a better way to do this?

**Review by icyphox — CHANGES_REQUESTED** [review]

File: plugins/link-handler.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+// NOTE(nojusr): provides a basic, short description of whatever is inside
+// a posted URL.
+func getDescriptionFromURL(url string) string {
+	resp, err := http.Get(url)
+
+	if err != nil {
+		return ""
+	}
```
Comment (icyphox) [inline plugins/link-handler.go]: pass the error up the stack.

**Review by icyphox — CHANGES_REQUESTED** [review]

File: plugins/plugins.go (hunk 3 above)
Comment (adedomin) [reply plugins/plugins.go]:
```
gore> :import "strings"
gore> strings.HasPrefix("fdsafdsafdsafdsa", "")
true
```
why the || trigger == "" ?

**Pull request merged**